test-integration/replication.test.js

```diff
@@ -32,36 +30,22 @@ const data = [
     uri: 'mongodb://localhost:27002'
   }
 ]
-const servers = []
-
-for (let index = 0; index < data.length; index += 1) {
-  const { hostname, uri } = data[index]
-  const client = new MongoClient(uri)
-  const server = new MongoServerState(hostname, client)
-  data[index].server = server
-  servers.push(server)
-}
+const servers = data.map((settings) => new ReplicaMemberMongoServerState(settings))
```

Comment:
Can't this conversion logic behandled in the `ReplicaSetState` this way the core lib and the cli can do the exact same thing? they just get called with  JSON/string content. 

The reason for this is want to keep this logic low down so it can be easily tested and reused between lib and cli.